lib/specinfra/ext/class.rb

```diff
@@ -2,7 +2,7 @@ class Class
   def subclasses
     result = []
     ObjectSpace.each_object(Class) do |k|
-      next if k.name.nil?
+      next unless k == k.ancestors.first #Skips if this is a singleton class
```

Comment:
@mizzy @biinari Thanks for input. It appears like Ruby may have introduced a backwards incompatibility with the way ancestors are determined for singleton_class between Ruby 2.0 and Ruby 2.1.

Source: https://gist.github.com/floehopper/8637991
```
# ruby 2.0.0p0 (2013-02-24 revision 39474) [x86_64-darwin12.2.0]
class X; end
X.ancestors # => [X, Object, Kernel, BasicObject]
X.singleton_class # => #<Class:X>
X.singleton_class.ancestors # => [Class, Module, Object, Kernel, BasicObject]

# ruby 2.1.0p0 (2013-12-25 revision 44422) [x86_64-darwin13.0]
class X; end
X.ancestors # => [X, Object, Kernel, BasicObject]
X.singleton_class # => #<Class:X>
X.singleton_class.ancestors # => [#<Class:X>, #<Class:Object>, #<Class:BasicObject>, Class, Module, Object, Kernel, BasicObject]

# The following are extra ancestors in Ruby v2.1
# [#<Class:X>, #<Class:Object>, #<Class:BasicObject>]
```

With this development, I would defer to using #560 as the proper fix, as I'd be perfectly happy defaulting to the old behavior on < Ruby 2.0

Thanks!